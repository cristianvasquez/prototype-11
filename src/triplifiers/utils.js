import {ns} from '../namespaces.js'

// Poor's man iterator
function * applyToChunks (text, func, char='\n') {
  if (text.indexOf(char) !== -1) {
    let i = 0
    let j = 0
    while ((j = text.indexOf(char, i)) !== -1) {
      const line = text.substring(i, j)
      yield func(line)
      i = j + 1
    }
  } else {
    yield func(text)
  }
}

function shrink (iri) {
  const candidates = Array.from(Object.entries(ns)).filter(([_,value]) => {
    return iri.startsWith(value().value)
  })
  if (candidates.length) {
    candidates.sort(([, iri1], [, iri2]) => iri2.length - iri1.length)
    const found = candidates[0]
    return  iri.replace(new RegExp(`^${found[1]().value}`), `${found[0]}:`)
  }
  return iri
}

export {applyToChunks, shrink}