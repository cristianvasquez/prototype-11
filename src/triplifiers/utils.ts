// Poor man's iterator
function* applyToChunks(text: string, func: any, char = '\n') {
    if (text.indexOf(char) !== -1) {
        let i = 0
        let j = 0
        while ((j = text.indexOf(char, i)) !== -1) {
            const line = text.substring(i, j)
            yield func(line)
            i = j + 1
        }
    } else {
        yield func(text)
    }
}

export {applyToChunks}