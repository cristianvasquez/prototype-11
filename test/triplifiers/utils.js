import { shrink,getPrefixes } from '../../src/triplifiers/utils.js'

import expect from 'expect'
import toMatchSnapshot from 'expect-mocha-snapshot'

expect.extend({ toMatchSnapshot })

describe('[utils]', function () {
  describe('[shrink]', function () {
    const phrases = [
      'http://vault.org/something',
      'http://www.w3.org/1999/02/22-rdf-syntax-ns#type',
      'http://www.not-me.com'
    ]
    phrases.forEach((current) => {
      it(`"${current}"`, function () {
        const actual = shrink(current)
        expect(actual).toMatchSnapshot(this)
      })
    })
  })

  describe('[prefixes]', function () {
    it(`"values"`, function () {
      const actual = getPrefixes()
      expect(actual).toMatchSnapshot(this)
    })
  })


})