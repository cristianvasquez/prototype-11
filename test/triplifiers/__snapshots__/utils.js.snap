// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[utils] [shrink] "http://vault.org/something" 1`] = `"vault:something"`;

exports[`[utils] [shrink] "http://www.not-me.com" 1`] = `"http://www.not-me.com"`;

exports[`[utils] [shrink] "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" 1`] = `"rdf:type"`;
